### PR TITLE
[TA] Hide StringIndexType

### LINKED ISSUE #21367: Hide StringIndexType from TextAnalyticsRequestOptions
Not everyone needs this property, so to avoid confusion, let's hide it.
e should still document that this capability exists for people that want to use it.

### PR DESCRIPTION
**Pull request by maririos**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/21367

### CODE REVIEW

File: sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs [hunk 1]
```diff
@@ -342,7 +342,7 @@ internal static EntityLinkingTask ConvertToLinkedEntitiesTask(RecognizeLinkedEnt
                 Parameters = new EntityLinkingTaskParameters()
                 {
                     ModelVersion = action.ModelVersion,
-                    StringIndexType = action.StringIndexType,
+                    StringIndexType = StringIndexType.Utf16CodeUnit,
```
Comment (deyaaeldeen) [inline sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs:345]: NIT: Let's move `_stringIndexType` outside the client class file and use it here too.

Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.cs]: Created `Constants` class

**Review by deyaaeldeen — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md [hunk 2]
```diff
@@ -21,6 +21,7 @@
 - Changed type `RecognizePiiEntitiesOptions.DomainFilter` from `PiiEntityDomainType?` to `PiiEntityDomainType`.
 - Changed type `AnalyzeActionsOptions.IncludeStatistics` from `bool` to `bool?`.
 - Removed property `Statistics` from `AnalyzeActionsResult` as it is not currently returned by the service even if the user passes `IncludeStatistics  = true`.
+- Removed property `StringIndexType` from `TextAnalyticsRequestOptions`.This SDK will keep using `UTF-16` code unit as the default encoding.
```
Comment (kinelski) [inline sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md:24]: nit: missing blank space before "This SDK..."

**Review by kinelski — APPROVED** [review]

File: sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md (hunk 2 above)
Comment (maririos) [reply sdk/textanalytics/Azure.AI.TextAnalytics/CHANGELOG.md]: Great eyes!

**Pull request merged**